assets/src/edit-story/components/panels/size.js

```diff
@@ -18,101 +18,216 @@
+      <Row expand>
+        <BoxedNumeric
+          prefix={__('W', 'web-stories')}
+          value={state.width}
+          isMultiple={width === ''}
+          onChange={(value) => {
+            const ratio = width / height;
+            const newWidth =
+              isNaN(value) || value === '' ? '' : parseFloat(value);
+            setState({
+              ...state,
+              width: newWidth,
+              height:
+                height !== '' && typeof newWidth === 'number' && lockRatio
+                  ? dataPixels(newWidth / ratio)
+                  : height,
+            });
+          }}
+          disabled={isFill}
+        />
+        <Toggle
+          icon={<Locked />}
+          uncheckedIcon={<Unlocked />}
+          value={lockRatio}
+          isMultiple={false}
+          onChange={(value) => {
+            setLockRatio(value);
+          }}
+          disabled={isFill}
+        />
+        <BoxedNumeric
+          prefix={__('H', 'web-stories')}
+          value={state.height}
+          isMultiple={height === ''}
+          onChange={(value) => {
+            const ratio = width / height;
+            const newHeight =
+              isNaN(value) || value === '' ? '' : parseFloat(value);
+            setState({
+              ...state,
+              height: newHeight,
+              width:
+                width !== '' && typeof newHeight === 'number' && lockRatio
+                  ? dataPixels(newHeight * ratio)
+                  : width,
+            });
+          }}
+          disabled={isFill}
+        />
+      </Row>
+      {/** Fill & Reset size */}
+      <Row expand={false} spaceBetween={false}>
+        <Toggle
+          icon={<Fullbleed />}
+          value={state.isFill}
+          isMultiple={false}
+          onChange={(value) => {
+            setState({
+              ...state,
+              isFill: value,
+            });
+          }}
+        />
+        {/** TODO: Implement size resetting */}
+        <Button onClick={() => {}}>{__('Reset size', 'web-stories')}</Button>
+      </Row>
+      {/** Rotation and Flipping */}
+      <Row expand={false} spaceBetween={false}>
+        <ExpandedNumeric
+          label={__('Rotate', 'web-stories')}
+          suffix={_x('°', 'Degrees, 0 - 360. ', 'web-stories')}
+          value={state.rotationAngle}
+          isMultiple={rotationAngle === ''}
+          onChange={(value) => {
+            setState({
+              ...state,
+              rotationAngle:
+                isNaN(value) || value === '' ? '' : parseFloat(value),
+            });
+          }}
+          disabled={isFill}
+        />
+        <Toggle
+          icon={<FlipHorizontal />}
+          value={/** TODO */ false}
+          isMultiple={false}
+          onChange={() => {}}
+        />
+        <Toggle
+          icon={<FlipVertical />}
+          value={/** TODO */ false}
+          isMultiple={false}
+          onChange={() => {}}
+        />
```

Comment:
I don't think these are thought to be toggles actually. It's just a button flipping the content, which you can press any number of times without the button changing appearance. I.e. I don't think it has a _"selected"_ state as it's never true or false. It's just a button. I might be wrong, though.

Reply:
Agreed. Even considering that we have a toggle UI proposal, it can create confusion, as you said, in the end, only the content is being replaced. I will investigate why it is using the Toggle component instead of the button. Nice catch!